### PR TITLE
TEC-5316 Fix TEC Tweaks compatibility

### PR DESCRIPTION
**Pull request by andrasguseo**
### 🎫 Ticket

[TEC-5316]

### 🗒️ Description

With TEC 6.5.0 we added code for compatibility with the TEC Tweaks extension. The code hooks into a filter with 2 params, but the filter only accepts one parameter, and as a result, throws a fatal error when visiting any Events > Settings page.

### 🎥 Artifacts 
[Screenshot the of fatal.](https://www.dropbox.com/scl/fi/afp7n8yn8xtn2nzz3pa8r/shot_241105_133205.jpg?rlkey=jogsddtmennrnqxkv5yx5u6sz&e=1&dl=0)
[Screenshot of the Tweaks settings page when fixed.](https://www.dropbox.com/scl/fi/a548uj990inlp1nacfa0w/shot_241105_133229.jpg?rlkey=6pec229ibnxelvv3hxtbad4yh&e=1&dl=0)

### ✔️ Checklist
- [x] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [ ] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [x] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

[TEC-5316]: https://stellarwp.atlassian.net/browse/TEC-5316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php [hunk 1]
```diff
@@ -39,21 +39,28 @@ protected function load(): void {
...
 	 */
 	public function remove_end_time_extension_settings() {
 		add_filter( 'tribe_get_option_tribe_ext_tec_tweaks_remove_event_end_time', '__return_empty_array' );
 		add_filter(
 			'tec_general_settings_viewing_section',
-			static function ( $fields, $id ) {
-				if ( $id !== 'tec-tweaks' ) {
+			static function ( $fields ) {
+				// Bail, if not on the Tweaks settings tab.
+				if ( $_REQUEST['tab'] !== 'tec-tweaks' ) {
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php:52]: @Camwyn Not sure how to better check if we are on the 'tec-tweaks' settings tab. Is there a better way than using `$_REQUEST`?

[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php:52]: @andrasguseo Use `tec_get_request_var`

[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php]: Thanks @rafsuntaskin

File: src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php [hunk 2]
```diff
@@ -39,21 +39,28 @@ protected function load(): void {
...
+
+				// Remove settings on the Tweaks tab.
+				// Compatibility for TEC Tweaks 1.1.1 and before.
 				unset( $fields['tec_labs_tec_tweaks_remove_event_end_time'] );
 
 				return $fields;
 			},
 			10,
-			2
+			1
```
Comment (rafsuntaskin) [inline src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php:63]: setting 1 is redundant also you can omit the 10 for priority as that is also default.

**Review by rafsuntaskin — APPROVED** [review]

File: src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php (hunk 2 above)
Comment (andrasguseo) [reply src/Events/Integrations/Plugins/TEC_Tweaks_Extension/Provider.php:63]: Noted. Thx!

**Pull request merged**